drivers/platform/mbed/mbed_gpio.cpp

```diff
@@ -0,0 +1,308 @@
+		/* Configure and instantiate GPIO pin as input */
+		gpio_input = new DigitalIn((PinName)desc->number);
+		if (!gpio_input)
+			return -ENOMEM;
+
+		gpio_desc_extra->gpio_pin = (struct mbed_gpio_desc *)gpio_input;
+		gpio_desc_extra->direction = GPIO_IN;
+
+		/* Set the gpio pin mode */
+		gpio_input->mode((PinMode)((struct mbed_gpio_desc *)desc->extra)->pin_mode);
+
+		return SUCCESS;
+	}
+
+	return -EINVAL;
+}
+
+/**
+ * @brief Enable the output direction of the specified GPIO.
+ * @param desc[in] - The GPIO descriptor.
+ * @param value[in] - The GPIO value.
+ *                Example: GPIO_HIGH
+ *                         GPIO_LOW
+ * @return SUCCESS in case of success, negative error code otherwise.
+ * @note does not support reconfiguration of already set pin direction
+ */
+int32_t mbed_gpio_direction_output(struct gpio_desc *desc, uint8_t value)
+{
+	DigitalOut *gpio_output;    	// pointer to gpio output object
+	struct mbed_gpio_desc
+		*gpio_desc_extra;   // pointer to gpio desc extra parameters
+
+	if (!desc|| !desc->extra)
+		return -EINVAL;
+
+	gpio_desc_extra = (struct mbed_gpio_desc *)(desc->extra);
+	if (gpio_desc_extra->gpio_pin == NULL) {
+		/* Configure and instantiate GPIO pin as output */
+		gpio_output = new DigitalOut((PinName)desc->number);
+		if (!gpio_output)
+			return -ENOMEM;
+
+		gpio_desc_extra->gpio_pin = (struct mbed_gpio_desc *)gpio_output;
+		gpio_desc_extra->direction = GPIO_OUT;
+
+		/* Set the GPIO value */
+		if (gpio_set_value(desc, value) != SUCCESS) {
+			free(gpio_output);
+			return -EIO;
+		}
+
+		return SUCCESS;
+	}
+
+	return -EINVAL;
+}
+
+/**
+ * @brief Get the direction of the specified GPIO.
+ * @param desc[in] - The GPIO descriptor.
+ * @param direction[in,out] - The GPIO direction.
+ *                    Example: GPIO_OUT
+ *                             GPIO_IN
+ * @return SUCCESS in case of success, negative error code otherwise.
+ */
+int32_t mbed_gpio_get_direction(struct gpio_desc *desc, uint8_t *direction)
+{
+	mbed_gpio_desc *gpio_desc_extra;     // pointer to gpio desc extra parameters
+
+	if (!desc || !desc->extra || !direction)
+		return -EINVAL;
+
+	gpio_desc_extra = (struct mbed_gpio_desc *)(desc->extra);
+
+	if (gpio_desc_extra->gpio_pin)
+		*direction = gpio_desc_extra->direction;
+
+	return SUCCESS;
+}
+
+/**
+ * @brief Set the value of the specified GPIO.
+ * @param desc[in] - The GPIO descriptor.
+ * @param value[in] - The GPIO value.
+ *                Example: GPIO_HIGH
+ *                         GPIO_LOW
+ * @return SUCCESS in case of success, negative error code otherwise.
+ */
+int32_t mbed_gpio_set_value(struct gpio_desc *desc, uint8_t value)
+{
+	DigitalOut *gpio_output; 		// pointer to gpio output object
+	mbed_gpio_desc *gpio_desc_extra;    // pointer to gpio desc extra parameters
+
+	if (!desc|| !desc->extra)
+		return -EINVAL;
+
+	gpio_desc_extra = (struct mbed_gpio_desc *)(desc->extra);
+
+	if (gpio_desc_extra->gpio_pin) {
```

Comment:
Is it worthwhile to get the direction first to make sure the pin is output? What happens if we try to write an input pin?

Reply:
Valid point. Added check for GPIO direction.